src/main/resources/application-qa.properties

```diff
@@ -0,0 +1,11 @@
+spring.datasource.qa.url=jdbc:h2:mem:testQa
```

Comment:
Seems you don't need to specify `qa` or `dev` in the path.
The appropriate file will be picked up from **profile** value. In the case of `qa` profile - only application-qa.properties file will be used.